src/App.js

```diff
@@ -14,48 +14,99 @@ const generateSquares = () => {
   for (let row = 0; row < 3; row += 1) {
     squares.push([]);
     for (let col = 0; col < 3; col += 1) {
-      squares[row].push({
-        id: currentId,
-        value: '',
-      });
+      squares[row].push({ id: currentId, value: "" });
       currentId += 1;
     }
   }
-
   return squares;
-}
+};
 
 const App = () => {
+  const [squares, setSquares] = useState(generateSquares()); // useState prevents this from being run every render
+  const [isPlayerOne, setIsPlayerOne] = useState(true);
+  const [filledSquares, setFilledSquares] = useState(0);
+  const [isWinner, setWinner] = useState();
+
+  // setSquares allows us to change the value of squares without manipulating it directly
 
-  const [squares, setSquares] = useState(generateSquares());
+  const onClickCallback = (id) => {
+    const newSquares = [...squares]; //clone of squares
+    if (isWinner || filledSquares === 9) {
+      return;
+    }
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+    for (let i = 0; i < 3; i++) {
+      for (let j = 0; j < 3; j++) {
+        if (newSquares[i][j].id === id && newSquares[i][j].value === "") {
+          // square clicked should be empty and there is no winner
+          newSquares[i][j].value = isPlayerOne ? PLAYER_1 : PLAYER_2;
+          setIsPlayerOne(!isPlayerOne);
+          setFilledSquares(filledSquares + 1);
+          setSquares(newSquares);
+          checkForWinner(squares);
+          return;
+        }
+      }
+    }
+  };
 
+  const allMySquaresValues = () => {
```

Comment:
I like that you pulled this out into a helper method!
Given that your solution lends itself to wanting to check the winner by looking at a 1-dimensional array, it would be an appropriate optimization to change the `generateSquares` function to just create a 1-D array from the beginning and use that as your state.
Truth be told, the main reason we set it up as a 2-D array is because we believed it would help many people with writing code to determine the winner.
But that's just an optimization! Not a significant problem with your code :)